src/status_im2/contexts/add_new_contact/events_test.cljs

```diff
@@ -10,47 +13,160 @@
 (def link-ckey (str "https://join.status.im/u/" ckey))
 (def link-ens (str "https://join.status.im/u/" ens))
 
-(deftest identify-type-test
-  (are [input expected] (= (core/identify-type input) expected)
-   ""        {:input    ""
-              :id       nil
-              :type     :empty
-              :ens-name nil}
-
-   ukey      {:input    ukey
-              :id       ukey
-              :type     :public-key
-              :ens-name nil}
-
-   ens       {:input    ens
-              :id       ens
-              :type     :ens-name
-              :ens-name ens-stateofus-eth}
-
-   ckey      {:input    ckey
-              :id       ckey
-              :type     :compressed-key
-              :ens-name nil}
-
-   link-ckey {:input    link-ckey
-              :id       ckey
-              :type     :compressed-key
-              :ens-name nil}
-
-   link-ens  {:input    link-ens
-              :id       ens
-              :type     :ens-name
-              :ens-name ens-stateofus-eth}))
-
-(deftest search-empty-string-test
-  (is (= (core/set-new-identity {:db {:contacts/new-identity :foo}} "")
-         {:db {}})))
-
-(deftest search-uncompressed-key-test
-  (is (= (core/set-new-identity {:db {}} ukey)
-         {:db {:contacts/new-identity
-               {:input      ukey
-                :public-key ukey
-                :state      :error
-                :error      :uncompressed-key}}})))
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; unit tests (validate-contact fn)
```

Comment:
do the deftest titles not make this clear enough?

Reply:
@J-Son89 The reason I did this, is because the first test is a smallest 'unit' test - it tests the functions that I moved as much of the analysis logic as I could into, so that I could have a lot of tests to cover all the permutations. The remaining tests are event handler tests, which require app-db setup and have some callbacks that are tested, and I liked having sections to clearly separate the two. I might be in the minority here (I guess @ilmotta likes them), but I typically use this practice to break up a ns like this when I'm conceptually thinking about it in sections like this. I figured someone might not like them, so I was prepared to remove them. Or maybe there's a better approach.  WDYT?

Reply:
ok got it. I think it's fine. I was more referring to the `validate-contact fn` part of this string but makes sense to leave and I don't have too strong an opinion with this. 👍